Source/HTTP/Curl/CurlProvider.cpp

```diff
@@ -0,0 +1,130 @@
+#include "pch.h"
+#include "CurlProvider.h"
+#include "CurlEasyRequest.h"
+
+using namespace xbox::http_client;
+
+HRESULT Internal_InitializeHttpPlatform(HCInitArgs* args, PerformEnv& performEnv) noexcept
+{
+    assert(args == nullptr);
+    UNREFERENCED_PARAMETER(args);
+
+    auto initResult = HC_PERFORM_ENV::Initialize();
+    RETURN_IF_FAILED(initResult.hr);
+
+    performEnv = initResult.ExtractPayload();
+
+    return S_OK;
+}
+
+void Internal_CleanupHttpPlatform(HC_PERFORM_ENV* performEnv) noexcept
+{
+    // HC_PERFORM_ENV created with custom deleter - HC_PERFORM_ENV needs to be destroyed and cleaned up explicitly.
+    http_stl_allocator<HC_PERFORM_ENV> a{};
+    std::allocator_traits<http_stl_allocator<HC_PERFORM_ENV>>::destroy(a, std::addressof(*performEnv));
+    std::allocator_traits<http_stl_allocator<HC_PERFORM_ENV>>::deallocate(a, performEnv, 1);
+}
+
+HRESULT Internal_SetGlobalProxy(
+    _In_ HC_PERFORM_ENV* performEnv,
+    _In_ const char* proxyUri
+) noexcept
+{
+    UNREFERENCED_PARAMETER(performEnv);
+    UNREFERENCED_PARAMETER(proxyUri);
+    return E_NOTIMPL;
+}
+
+void CALLBACK Internal_HCHttpCallPerformAsync(
+    _In_ HCCallHandle call,
+    _Inout_ XAsyncBlock* asyncBlock,
+    _In_opt_ void* context,
+    _In_ HCPerformEnv env
+) noexcept
+{
+    UNREFERENCED_PARAMETER(context);
+    assert(env);
+
+    HRESULT hr = env->Perform(call, asyncBlock);
+    if (FAILED(hr))
+    {
+        // Complete XAsyncBlock if we fail synchronously
+        XAsyncComplete(asyncBlock, hr, 0);
+    }
+}
+
+namespace xbox
+{
+namespace http_client
+{
+
+HRESULT HrFromCurle(CURLcode c) noexcept
+{
+    if (c == CURLE_OK)
+    {
+        return S_OK;
+    }
+    else
+    {
+        return E_FAIL;
```

Comment:
this is unfortunate; is there a way to expose the HRESULT from the OS_ERRNO when possible?

Reply:
There probably is.  Will think about that and update accordingly

Reply:
This is only getting used in cases where some curl API fails unexpectedly (ex curl_easy_setopt, curl_initialize)